### DISCUSSION TITLE
How can local compilation make debugging more convenient?

### DISCUSSION THREAD

**Discussion by SF-Simon** [dotnet/maui#13985] [Q&A]
Hello, everyone, I downloaded the main version to debug locally. After I modified the code, I used the command to complete the compilation and debugging. This is no problem.

```
$ dotnet tool restore
$ dotnet cake --sln="<download_directory>\MauiApp2\MauiApp2.sln" --target=VS --pack
```

But I modified the MAUI code again, and I don't know how to debug the new code. It's really troublesome to repeatedly execute the above code every time. 

Did I not master the skill? Can't there be hot update or other methods?

Please help me simplify this process, thank you.

**Comment by behdad088** [comment]
@SF-Simon  If you want to avoid the repetitive process of restoring tools, compiling and debugging every time you make a change to the code, there are several approaches you can take:

Use a development environment that supports hot reload:
Some development environments such as Visual Studio and Visual Studio Code support hot reload, which allows you to see the changes you make to your code without having to rebuild and restart your application. This can be a great time saver when you are making small changes and need to quickly see the results.

Use a watch command to monitor changes to the code:
You can use the dotnet watch command to monitor changes to your code and automatically rebuild and restart your application. For example, you can use the following command to start watching for changes to your code:

$ dotnet watch run --project <path-to-your-project

This command will automatically rebuild and restart your application every time you make a change to your code.

Use a Continuous Integration/Continuous Deployment (CI/CD) system:
A CI/CD system can automate the process of building, testing, and deploying your code every time you make a change. This can save you a lot of time and effort, especially if you are working on a large project with many contributors.

Use a tool like nodemon or supervisor:
These tools are designed to monitor changes to your code and automatically restart your application. While they are primarily used in the JavaScript world, you may be able to adapt them to work with your .NET application.

**Reply by Xyncgas** [reply]
Restarting your application doesn't hurt, you can do chores and eat and wait for the app to relaunch, realistically speaking it's better than having nothing to write, after a while you will get used to it, you can use the waiting to figure out other bugs or problems, take it easy
